### DISCUSSION TITLE
trace_ID confusion during concurrency

### DISCUSSION THREAD

**Discussion by lihaopeng1** [open-telemetry/opentelemetry-cpp#1928] [Q&A]
`namespace trace     	= opentelemetry::trace;`
`namespace nostd     	= opentelemetry::nostd;`
`nostd::shared_ptr<trace::Tracer> UJaeger::get_tracer()`
`{`
 `  auto provider = trace::Provider::GetTracerProvider();`
 `  auto tracer = provider->GetTracer(mod_name.c_str());`
 `  return tracer;`
`}`
`span = get_tracer()->StartSpan` 

Why do I have the same trace_ID when I am concurrent, I want each thread to have an independent trace_ID, 
Initial start() I hope `get_tracer()->GetCurrentSpan()->GetContext().trace_id();` to be 0
what should I do?

**Comment by marcalff** [comment]
"Why do I have the same trace_ID when I am concurrent" ... please clarify which trace_id() value is seen here, always 0 or always something that looks populated (and identical for all threads) ?

Please clarify how `StartSpanOptions` is passed to `StartSpan()`, and how the parent member is populated.

To start a root span, do not set a parent.

To start a child span, set the parent to the parent span.

Even for a root span, the trace_id should never be 0, it should be a random number for the new trace, and yes, separate traces for separate threads should have different non 0 trace_id(), generated by the call to StartSpan().

It would help to see actual values with an example, to investigate further.

**Comment by lihaopeng1** [comment]
What should I do if I want different requests in the same thread space to be buried as root span instead of active_span?
Due to business needs, our service does not perform thread isolation when burying points. When terminating, it does not end burying points according to the order of burying points, but terminates according to the span object returned by the burying points, so how should I avoid active_span? That is, the same thread can generate different root span.

**Comment by lalitb** [comment]
Moving this to discussion for now. We can move this back to issue list if this is valid bug.

@lihaopeng1  Your explanation is not very clear to me. Can you please try explaining your scenario once again.
